### PR TITLE
rework resource cleanup on Azure

### PR DESCRIPTION
**Pull request by jepio**
# rework resource cleanup on Azure

Rework cleanup of NIC/IP/disk resources on Azure to make it more reliable and a responsibility of the Azure Resource Manager. Thanks to this, OS disks are now deleted immediately with the VM. Also implement correct architecture detection on Azure.

## How to use

`kola run`

## Testing done

`kola run` completes slightly faster and tests find the correct `kolet` binary.

- [X] Changelog entries added in the respective `changelog/` directory (user-facing change, bug fix, security fix, update)

### CODE REVIEW

**Comment by jepio** [comment]
@tormath1 this is a followup to #270.

File: platform/api/azure/instance.go [hunk 1]
```diff
@@ -29,6 +29,8 @@ import (
 	"github.com/flatcar-linux/mantle/util"
 )
 
+var forceDelete = true
```
Comment (tormath1) [inline platform/api/azure/instance.go:32]: I think it can be safely defined as a constant:
```suggestion
const forceDelete = true
```

Comment (jepio) [reply platform/api/azure/instance.go:32]: Tried that, it's not legal to take the address of a constant this way

**Review by tormath1 — APPROVED** [review]
thanks for the cleanup :racing_car:

File: kola/harness.go [hunk 2]
```diff
@@ -577,6 +577,9 @@ func architecture(pltfrm string) string {
 	if pltfrm == "aws" && AWSOptions.Board != "" {
 		nativeArch = boardToArch(AWSOptions.Board)
 	}
+	if pltfrm == "azure" && AzureOptions.Board != "" {
```
Comment (krnowak) [inline kola/harness.go:580]: Oh, the saving of keystrokes. Extending lifetime of our dear keyboards. Heh.

(No need to change anything, just being amused here. :) )

**Review by krnowak — APPROVED** [review]

File: kola/harness.go (hunk 2 above)
Comment (jepio) [reply kola/harness.go:580]: I think it's still too long, the `r` is a waste of space

**Pull request merged**